docs/bigtable-data-api.rst

```diff
@@ -0,0 +1,328 @@
+
+.. note::
+
+    If ``state`` is passed when no ``filter_`` is set on a
+    :class:`Row <gcloud.bigtable.row.Row>`, adding the mutation will fail.
+    Similarly, if no ``state`` is passed when a ``filter_`` has been set,
+    adding the mutation will fail.
+
+Append Mutations
+----------------
+
+Append mutations can be added via one of two methods
+
+* :meth:`append_cell_value <gcloud.bigtable.row.Row.append_cell_value>` appends
+  a bytes value to an existing cell:
+
+  .. code:: python
+
+      row.append_cell_value(column_family_id, column, bytes_value)
+
+* :meth:`increment_cell_value <gcloud.bigtable.row.Row.increment_cell_value>` increments
+  an integer value in an existing cell:
+
+  .. code:: python
+
+      row.increment_cell_value(column_family_id, column, int_value)
+
+  Since only bytes are stored in a cell, the cell value is decoded as
+  an unsigned 64-bit integer before being incremented. (This happens on
+  the Google Cloud Bigtable server, not in the library.)
+
+Notice that no timestamp was specified. This is because **append** mutations
+operate on the latest value of the specified column.
+
+If there are no cells in the specified column, then the empty string (bytes
+case) or zero (integer case) are the assumed values.
+
+Starting Fresh
+--------------
+
+If accumulated mutations need to be dropped, use
+:meth:`clear_mutations() <gcloud.bigtable.row.Row.clear_mutations>`
+
+.. code:: python
+
+    row.clear_mutations()
+
+To clear **append** mutations, use
+:meth:`clear_modification_rules() <gcloud.bigtable.row.Row.clear_modification_rules>`
+
+.. code:: python
+
+    row.clear_modification_rules()
+
+Reading Data
+++++++++++++
+
+Read Single Row from a Table
+----------------------------
+
+To make a `ReadRows`_ API request for a single row key, use
+:meth:`Table.read_row() <gcloud.bigtable.table.Table.read_row>`:
+
+.. code:: python
+
+    row_data = table.read_row(row_key)
+
+Rather than returning a :class:`Row <gcloud.bigtable.row.Row>`, this method
+returns a :class:`PartialRowData <gcloud.bigtable.row_data.PartialRowData>`
+instance. This class is used for reading and parsing data rather than for
+modifying data (as :class:`Row <gcloud.bigtable.row.Row>` is).
+
+A filter can also be applied to the
+
+.. code:: python
+
+    row_data = table.read_row(row_key, filter_=filter_val)
+
+The allowable ``filter_`` values are the same as those used for a
+:class:`Row <gcloud.bigtable.row.Row>` with **conditional** mutations. For
+more information, see the
+:meth:`Table.read_row() <gcloud.bigtable.table.Table.read_row>` documentation.
+
+Stream Many Rows from a Table
+-----------------------------
+
+To make a `ReadRows`_ API request for a stream of rows, use
+:meth:`Table.read_rows() <gcloud.bigtable.table.Table.read_rows>`:
+
+.. code:: python
+
+    row_data = table.read_rows()
+
+Using gRPC over HTTP/2, a continual stream of responses will be delivered.
+We have a custom
```

Comment:
The sentence plays weird to me. It seems to imply that there's a different `PartialRowsData` class at play but that doesn't appear to be the case. I think it's better to just remove this.